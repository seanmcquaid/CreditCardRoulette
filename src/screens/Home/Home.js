import React from "react";
import {View, Text, Button} from "react-native";
import styled from "styled-components/native";

const Home = ({navigation}) => (
    <HomeScreenContainer>
        <Text>Credit Card Roulette</Text>
        <Button title="Play" onPress={() => navigation.push("Enter Check")}/>
    </HomeScreenContainer>
);

const HomeScreenContainer = styled(View)`
    flex : 1;
    justify-content : center;
    align-items : center;
`;


export default Home;