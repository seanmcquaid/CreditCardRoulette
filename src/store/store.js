import React, {createContext, useReducer} from "react";
import rootReducer from "../reducers/rootReducer";

const initialState = {
    checkAmount : 0,
    playerNames : []
};

export const GlobalContext = createContext();

export const GlobalContextProvider = props => {
    const [state, dispatch] = useReducer(rootReducer, initialState);

    return (
        <GlobalContext.Provider value={[state, dispatch]}>
            {props.children}
        </GlobalContext.Provider>
    )
};
