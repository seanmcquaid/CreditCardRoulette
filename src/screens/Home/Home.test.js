import React from "react";
import {GlobalContextProvider} from "../../store/store";
import {render, fireEvent} from "react-native-testing-library";
import Home from "./Home";

describe("<Home/>", () => {
    it("Button onPress works correctly", async () => {
        const navigate = jest.fn();
        
        const props = {
            navigation : {
                navigate
            }
        };

        const {getByTestId} = render(
            <GlobalContextProvider>
                <Home {...props}/>
            </GlobalContextProvider>);
        
        await fireEvent.press(getByTestId("playButton"));

        expect(navigate).toHaveBeenCalledTimes(1);
        expect(navigate).toHaveBeenCalledWith("Enter Check");
    });
    it("Matches snapshot correctly", () => {
        const navigate = jest.fn();
        
        const props = {
            navigation : {
                navigate
            }
        };

        const home = render(
            <GlobalContextProvider>
                <Home {...props}/>
            </GlobalContextProvider>);

        expect(home).toMatchSnapshot();
    });
});