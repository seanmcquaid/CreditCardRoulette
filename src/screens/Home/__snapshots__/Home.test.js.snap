// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Matches snapshot correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#00ae6b",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Fira-Sans",
          "fontSize": 26,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
        ]
      }
    >
      Credit 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#C5B358",
          },
        ]
      }
    >
      Card 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#B22222",
          },
        ]
      }
    >
      Roulette
    </Text>
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#C5B358",
          "borderColor": "#000",
          "borderRadius": 3,
          "borderStyle": "solid",
          "borderWidth": 2,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 16,
          "paddingBottom": 16,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 16,
        },
      ]
    }
    testID="button"
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Merriweather-Regular",
            "fontSize": 16,
          },
          null,
        ]
      }
      testID="buttonTitle"
    >
      Play
    </Text>
  </View>
</View>
`;
